### PR TITLE
fix: add exception handling to rendering

### LINKED ISSUE #314: feat: add error catch in all render methods
**Is your feature request related to a problem? Please describe.**
Yes, the getStaticProps and getServerProps code is not error handles and results in server crashes.

**Describe the solution you'd like**
Add a try/catch error handler in all Nextjs render methods
- getStaticPaths
- getStaticProps
- getServerSideProps

For an error case, the error should be passed in the props instead and should be handled in the page.

### PR DESCRIPTION
**Pull request by Shurtu-gal**
### Your checklist for this pull request

🚨Please review the [guidelines for contributing](https://github.com/Monday-Morning/project-tahiti/blob/main/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch** (left side). Also you should start _your branch_ off _default branch_.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [ ] I have added necessary documentation (if appropriate)

### Linked Issues
Fixes #314

### Description

Please describe your pull request.

❤️Thank you!

### Post merge checklist

- [x] Follow steps from the [guidelines for contributing](https://github.com/Monday-Morning/project-tahiti/blob/main/CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

**Comment by 120EE0692** [comment]
@Shurtu-gal this will show 404 page, it would be better if we can get 500 page.

**Comment by 120EE0692** [comment]
@Shurtu-gal 
I think this will redirect to page [404](https://mondaymorning.nitrkl.ac.in/404) even in case of error.
Please make sure in case of some error it redirects to page [500](https://mondaymorning.nitrkl.ac.in/500)

File: client/src/pages/archive/[...yearAndMonth].jsx [hunk 1]
```diff
@@ -102,40 +107,56 @@ export async function getStaticProps({
...
+    });
+    return {
+      props: { archiveArticles, year, month },
+      revalidate: 7 * 24 * 60 * 60,
+    };
+  } catch (err) {
+    return {
+      props: {
+        isError: true,
+        // error: err,
```
Comment (120EE0692) [inline client/src/pages/archive/[...yearAndMonth].jsx:130]: @Shurtu-gal pleasse remove this comment. This addes no value to the code and it's readability. Do this is all the files.

**Review by 120EE0692 — CHANGES_REQUESTED** [review]

File: client/src/pages/article/[...article].jsx [hunk 2]
```diff
@@ -143,44 +170,33 @@ export async function getStaticProps({
...
-  if (articleSlug !== getArticleSlug(article.title)) {
+  } catch (err) {
     return {
-      redirect: {
-        destination: getArticleLink(articleId, article.title),
-        permanent: false,
+      props: {
+        isError: true,
       },
+      notFound: true,
```
Comment (120EE0692) [inline client/src/pages/article/[...article].jsx:197]: @Shurtu-gal why notFound?

File: client/src/pages/index.jsx [hunk 3]
```diff
@@ -92,38 +92,48 @@ function HomePage({ issues, squiggles }) {
...
+          ? 60 * 60 * 1
+          : 60 * 60 * 24 * 2, // 1 Hour or 2 Days
+    };
+  } catch (err) {
+    return {
+      props: {
+        isError: true,
+        error: err,
+      },
+      notFound: true,
```
Comment (120EE0692) [inline client/src/pages/index.jsx:134]: @Shurtu-gal why notFound and error used here?

**Review by 120EE0692 — CHANGES_REQUESTED** [review]

File: client/src/pages/expressions/[...subCategory].jsx [hunk 4]
```diff
@@ -112,7 +118,7 @@ const EditorialPage = ({
           content='Monday Morning is the Media Body of National Institute Of Technology Rourkela. Monday Morning covers all the events, issues and activities going on inside the campus. Monday morning also serves as a link between the administration and the students.'
         />
       </Head>
-      {!isLoading && !isFallback ? (
+      {!isLoading && !isError && !isFallback ? (
```
Comment (120EE0692) [inline client/src/pages/expressions/[...subCategory].jsx:121]: @Shurtu-gal is this change required?

**Review by 120EE0692 — APPROVED** [review]

File: client/src/pages/expressions/[...subCategory].jsx [hunk 5]
```diff
@@ -138,50 +144,69 @@ export async function getStaticProps({
...
-    },
-    revalidate:
-      preview || new Date(Date.now()).getDay() < 3
-        ? 60 * 60 * 1
-        : 60 * 60 * 24 * 2, // 1 Hour or 2 Days
-  };
+  try {
+    if (subCategory == 'length') {
+      throw 'Paths not found';
+    }
```
Comment (rutajdash) [inline client/src/pages/expressions/[...subCategory].jsx:150]: Why this block?

**Review by rutajdash — COMMENTED** [review]
Would it be possible to use the 500 error inside the header/head tags? We don't want any page to go without having some headers being passed.

**Comment by Shurtu-gal** [comment]
> Would it be possible to use the 500 error inside the header/head tags? We don't want any page to go without having some headers being passed.

Yes, it's done.

**Review by rutajdash — APPROVED** [review]
Custom error boundaries left for future scope.

**Comment by rutajdash** [comment]
To be merged soon.

**Review by rutajdash — CHANGES_REQUESTED** [review]
image

Please fix these issues

**Comment by 120EE0692** [comment]
> image
> 
> Please fix these issues

@Shurtu-gal if there is no prettier issue then do the following steps
1) rename the file 500.jsx to something else, commit and push the changes.
2) then rename it back to 500.jsx, then commit and push.

**Review by rutajdash — APPROVED** [review]

**Pull request merged**